### DISCUSSION TITLE
Market Data Source Agnostic

### DISCUSSION THREAD

**Discussion by alfredo-catalano** [tarkah/tickrs#88] [Ideas]
Any plans to provide ability to connect to other market data feeds?
Data source agnostic? => https://crates.io/crates/fin_model

**Comment by tarkah** [comment]
Cool crate! I actually used a completely different data source that was free before they locked down the API, and then I moved to Yahoo Finance. The API migration was actually fairly pain free as OHCLV data is super easy to standardize and most likely very similar already. So I think a "generic" data model would be very easy to migrate to.

That said, I don't plan on supporting more than Yahoo Finance and I never plan to support any non-free datasources. This is a hobby project for me and it's just not how I plan to spend my time. This tool was never intended to be something for in-depth, critical analysis and I feel the Yahoo data, though sometimes a little behind (like options data), fits the bill perfectly for a program that's just intended to keep an eye on things.

**Reply by cwarden** [reply]
Yahoo Finance API is [mostly available](https://stackoverflow.com/questions/76059562/yahoo-finance-api-get-quotes-returns-invalid-cookie) now.  Any interest in keeping the project alive with a different data source?

**Reply by cwarden** [reply]
Looks like I'm mistaken and Yahoo Finance API lives on with 1c20bcdcb14ba222bfcbf5df90669cbb2bbec108.

**Comment by alfredo-catalano** [comment]
Any of the below would be awesome: 

Interactive Brokers
https://www.alphavantage.co/
https://developer.tradier.com/
